src/plugins/walletConnectToDapps/components/modals/SessionProposal.tsx

```diff
@@ -180,29 +181,68 @@ const SessionProposal = forwardRef<SessionProposalRef, WalletConnectSessionModal
       )
     }, [accountIdsByChainId, optionalNamespaces, requiredNamespaces, wallet])
 
-    const approvalNamespaces: SessionTypes.Namespaces = useMemo(
-      () => createApprovalNamespaces(requiredNamespaces, optionalNamespaces, selectedAccountIds),
-      [optionalNamespaces, requiredNamespaces, selectedAccountIds],
+    const handleConnectAccountIds = useCallback(
+      async (_selectedAccountIds: AccountId[]) => {
+        const approvalNamespaces: SessionTypes.Namespaces = createApprovalNamespaces(
+          requiredNamespaces,
+          optionalNamespaces,
+          _selectedAccountIds,
+        )
+
+        // exit if the proposal was not found - likely duplicate call rerendering shenanigans
+        const pendingProposals = web3wallet.getPendingSessionProposals()
+        if (
+          !Object.values(pendingProposals).some(
+            pendingProposal => pendingProposal.id === proposal.id,
+          )
+        ) {
+          return
+        }
+
+        setIsLoading(true)
+
+        const session = await web3wallet.approveSession({
+          id: proposal.id,
+          namespaces: approvalNamespaces,
+        })
+        dispatch({ type: WalletConnectActionType.ADD_SESSION, payload: session })
+        handleClose()
+      },
+      [dispatch, handleClose, proposal, web3wallet, optionalNamespaces, requiredNamespaces],
     )
 
-    const handleApprove = useCallback(async () => {
-      // exit if the proposal was not found - likely duplicate call rerendering shenanigans
-      const pendingProposals = web3wallet.getPendingSessionProposals()
-      if (
-        !Object.values(pendingProposals).some(pendingProposal => pendingProposal.id === proposal.id)
-      ) {
-        return
-      }
+    const handleConnectSelectedAccountIds = useCallback(
+      () => handleConnectAccountIds(selectedAccountIds),
+      [handleConnectAccountIds, selectedAccountIds],
+    )
 
-      setIsLoading(true)
+    const handleConnectAll = useCallback(() => {
+      const namespacesChainIds = Object.values({
+        ...requiredNamespaces,
+        ...optionalNamespaces,
+      }).flatMap(namespace => namespace.chains ?? [])
 
-      const session = await web3wallet.approveSession({
-        id: proposal.id,
-        namespaces: approvalNamespaces,
+      const filteredAccountIds = portfolioAccountIds.filter(accountId => {
```

Comment:
Technically, we could just select all EVM AccountIds, but could handle with bad states re: supported chains. Better off ensuring we select only what's supported to avoid errors.